src/main/java/org/broadinstitute/hellbender/tools/walkers/sv/SVAnnotate.java

```diff
@@ -0,0 +1,479 @@
+                updateVariantConsequenceDict(variantConsequenceDict, GATKSVVCFConstants.PROMOTER, promoter.getName());
+                anyPromoterOverlaps = true;
+            }
+        }
+        return anyPromoterOverlaps;
+    }
+
+    private void annotateNonCodingOverlaps(final SimpleInterval variantInterval, final Map<String, Set<String>> variantConsequenceDict) {
+        final Set<FullBEDFeature> nonCodingFeaturesForVariant = nonCodingOverlapDetector.getOverlaps(variantInterval);
+        for (FullBEDFeature feature : nonCodingFeaturesForVariant) {
+            String consequence = GATKSVVCFConstants.NONCODING_BREAKPOINT;
+            if (variantSpansFeature(variantInterval, new SimpleInterval(feature))) {
+                consequence = GATKSVVCFConstants.NONCODING_SPAN;
+            }
+            updateVariantConsequenceDict(variantConsequenceDict, consequence, feature.getName());
+        }
+    }
+
+
+    protected static void annotateNearestTranscriptionStartSite(final SimpleInterval variantInterval, final Map<String, Set<String>> variantConsequenceDict, SVIntervalTree<String> transcriptionStartSiteTree, int maxContigLength, int variantContigID) {
+        // TODO: keep all nearest TSS for dispersed CPX / CTX or choose closest?
+        // TODO: will start < end ever? Shouldn't at this point in the pipeline
+        SVIntervalTree.Entry<String> nearestBefore = transcriptionStartSiteTree.max(new SVInterval(variantContigID, variantInterval.getStart(), variantInterval.getEnd()));
+        SVIntervalTree.Entry<String> nearestAfter = transcriptionStartSiteTree.min(new SVInterval(variantContigID, variantInterval.getStart(), variantInterval.getEnd()));
+        // nearest TSS only "valid" for annotation if non-null and on the same contig as the variant
+        boolean beforeInvalid = (isNull(nearestBefore) || nearestBefore.getInterval().getContig() != variantContigID );
+        boolean afterInvalid = (isNull(nearestAfter) || nearestAfter.getInterval().getContig() != variantContigID );
+        // if at least one result is valid, keep one with shorter distance
+        if (!(beforeInvalid && afterInvalid)) {
+            // if result is invalid, set distance to longest contig length so that other TSS will be kept
+            int distanceBefore = beforeInvalid ? maxContigLength : variantInterval.getStart() - nearestBefore.getInterval().getEnd();
+            int distanceAfter = afterInvalid ? maxContigLength : nearestAfter.getInterval().getStart() - variantInterval.getEnd();
+            String nearestTSSGeneName = (distanceBefore < distanceAfter) ? nearestBefore.getValue() : nearestAfter.getValue();
+            updateVariantConsequenceDict(variantConsequenceDict, GATKSVVCFConstants.NEAREST_TSS, nearestTSSGeneName);
+        }
+        // TODO: return consequence instead?
+    }
+
+    private StructuralVariantAnnotationType getSVType(final VariantContext variant) {
```

Comment:
Would it be simpler to use the SVTYPE info field rather than the ALT allele?  Or does that not handle all the cases?